azext_iot/tests/test_iot_digitaltwin_unit.py

```diff
@@ -364,51 +386,162 @@ def test_iot_digitaltwin_property_update_error(self, fixture_cmd, serviceclient_
+                                               source_model='private', repo_id=mock_pnptarget['repository_id'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               interface='environmentalSensor',
+                                               source_model='device', repo_login=mock_pnptarget['cs'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               source_model='public',
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               source_model='private', repo_id=mock_pnptarget['repository_id'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               source_model='device', repo_login=mock_pnptarget['cs'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(source_model='public',
+                                               login=mock_target['cs']))])
+    def test_iot_digitaltwin_monitor_events(self, fixture_cmd, serviceclient, req):
+        subject.iot_digitaltwin_monitor_events(fixture_cmd,
+                                               req['device_id'],
+                                               req['device_query'],
+                                               req['interface'],
+                                               req['source_model'],
+                                               req['repo_endpoint'],
+                                               req['repo_id'],
+                                               req['consumer_group'],
+                                               req['timeout'],
+                                               req['hub_name'],
+                                               req['resource_group_name'],
+                                               req['yes'],
+                                               req['properties'],
+                                               req['repair'],
+                                               req['login'],
+                                               req['repo_login'])
+        if req['device_id']:
```

Comment:
Now that the `fixture_monitor_events` is here, we should assert it is receiving the proper inputs based on the request. For example when a device id is not input, then we assert it is not passed into _iot_hub_monitor_events. _iot_hub_monitor_events is essentially the interface to monitoring activities and we need to make sure its receiving the right inputs in all the cases.

Reply:
fixed